src/+otp/+kortewegdevries/KortewegdeVriesProblem.m

```diff
@@ -0,0 +1,100 @@
+classdef KortewegdeVriesProblem < otp.Problem
+    % Describes the evolution of waves on shallow water surfaces.
+    % This problem is discretized on a periodic domain.
+    %
+    % u_t = a (th (u^2)_x + (1 - th)(2u) u_x) + r u_x + n u_xxx 
+    %
+    % Ascher, Uri M. and Robert I. McLachlan. “On Symplectic and 
+    % Multisymplectic Schemes for the KdV Equation.” 
+    % Journal of Scientific Computing 25 (2005): 83-104.
+    %
+    % Robert M. Miura, Clifford S. Gardner, Martin D. Kruskal;
+    % Korteweg‐de Vries Equation and Generalizations. II. 
+    % Existence of Conservation Laws and Constants of Motion. 
+    % J. Math. Phys. 1 August 1968; 9 (8): 1204–1209. 
+    % https://doi.org/10.1063/1.1664701
+
+    methods
+        function obj = KortewegdeVriesProblem(timeSpan, y0, parameters)
+            
+            obj@otp.Problem('Korteweg de Vries', [], ...
+                timeSpan, y0, parameters);
+            
+        end
+    end
+    
+    properties (SetAccess = private)
+        Grid
+        DistanceFunction
+        NumInvariants
+        Invariant
```

Comment:
I wonder if this could be an RHS property? @AndreyAPopov

Reply:
The problem is that we will have other equations where invariants will be part of the RHS. We might want to just standardize that it belongs in RHS.

For instance, how will you handle Double Pendulum?

Reply:
I feel like the invariant is part of the RHS, as it requires the RHS parameters. 

In other problems where we have a distancefunction, that distance only depends on the discretization, not on how the RHS is constructed. Here, the invariants are actually a function of the RHS, thus I think this might belong there.

Reply:
Yeah this makes sense to me. The Invariant isn't intended to be used by a DAE solver (although it could be)

Reply:
So do you prefer this in the RHS ? I think the constraints that are used for solving the ODE should be in the RHS. Those that are not used for solution shouldn't be in the RHS. They should just be standalone functions. 
When I want to look at invariants in my file, it feels wrong to use RHS.invariant()

Reply:
The thing is, what you are defining is NOT an invariant of the system. It is the discrete invariant, which is tied uniquely to how you define the RHS. If you used different derivative operators you would get a different invariant. 

Your invariant is inherently TIED to how the system is solved.

I will let Steven decide ultimately, but that is my argument.

Reply:
It does depend on the RHS is the parametric sense(alpha, nu, rho, discretization). With the same line of thought, isn't the distance function also dependent on the discretization. Ultimately, I guess it depend on how you philosophically see the RHS class. To me, RHS is only functions that help me solve the ODE/PDE. But if you want it in RHS, I'll do it